redfish-core/lib/pcie_slots.hpp

```diff
@@ -139,13 +139,68 @@ inline void
             "xyz.openbmc_project.Inventory.Item.PCIeDevice"});
 }
 
+inline void
+    linkAssociatedProcessor(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& pcieSlotPath, size_t index)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, pcieSlotPath,
+         index](const boost::system::error_code ec,
+                const std::variant<std::vector<std::string>>& endpoints) {
+            if (ec)
+            {
+                if (ec.value() == EBADR)
+                {
+                    // This PCIeSlot have no processor association.
+                    BMCWEB_LOG_DEBUG << "No processor association found";
+                    return;
+                }
+                BMCWEB_LOG_ERROR << "DBUS response error" << ec.message();
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            const std::vector<std::string>* value =
+                std::get_if<std::vector<std::string>>(&(endpoints));
+
+            if (value == nullptr)
+            {
+                BMCWEB_LOG_DEBUG << "Error getting processor association!";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            const std::vector<std::string> processorPath = *value;
+            const std::string& cpuName =
```

Comment:
I would have expected:

`std::string cpuName = sdbusplus::message::object_path(processorPath[0]).filename();
`

this is a little strange, and I didn't see this pattern in upstream bmcweb.  Why are we binding these temporaries to const references?

Reply:
Hmm, agree.